core/psh/psh.c

```diff
@@ -89,15 +91,25 @@ int main(int argc, char **argv)
 	while (write(1, "", 0) < 0)
 		usleep(50000);
 
-	base = basename(argv[0]);
+	/* Check if its first shell */
+	pid = tcgetpgrp(STDIN_FILENO);
 
-	/* Run app */
-	if ((app = psh_findapp(base)) != NULL)
-		app->run(argc, argv);
-	else
-		fprintf(stderr, "psh: %s: unknown command\n", argv[0]);
+	do {
+		/* login prompt */
+		base = basename(argv[0]);
+		if (strcmp(base, "pshlogin") == 0 && (app = psh_findapp("auth")) != NULL)
+			while (app->run(0, NULL) != 1)
+				;
+
+		/* Run app */
+		if ((app = psh_findapp(base)) != NULL)
+			app->run(argc, argv);
+		else
+			fprintf(stderr, "psh: %s: unknown command\n", argv[0]);
```

Comment:
this should probably break the loop

Reply:
Yes, you are right. `break` added. Thanks